server/services/userService.ts

```diff
@@ -1,7 +1,7 @@
-import { convertToTitleCase } from '../utils/utils'
 import type HmppsAuthClient from '../data/hmppsAuthClient'
+import { convertToTitleCase } from '../utils/utils'
 
-interface UserDetails {
+export interface UserDetails {
```

Comment:
We'd like to use this in our new tests but I wonder if we should be using a factory instead. Though a factory for `users` felt right as it was a resource, I was less certain that a factory for an object such as `userDetails` worked. 

[Confusingly there is already an existing share type of the same name](https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-ui/blob/02393e7de798a6b780e6c6480da0145bda0701a2/server/%40types/ui/index.d.ts#L87) which requires has all 4 fields (including `roles`). I don't think this works for us as we want to test that the setting of roles is done.

Reply:
Not a typescript expert, but I think it makes sense to re-use this as a type rather than create a factory?
It seems like the `UserDetails` in `userService` is referring to a different thing than the shared `UserDetails` with the `roles` key? The `getUser` function wouldn't return an object with `roles`?